src/main/java/org/terracotta/statistics/GeneralOperationStatistic.java

```diff
@@ -53,26 +53,26 @@
    */
   @Override
   public long count(T type) {
-    return counts.get(type).sum();
+    return counts[type.ordinal()].sum();
   }
 
   @Override
   public long sum(Set<T> types) {
     long sum = 0;
     for (T t : types) {
-      sum += counts.get(t).sum();
+      sum += counts[t.ordinal()].sum();
     }
     return sum;
   }
 
   @Override
   public void end(T result) {
-    counts.get(result).increment();
+    counts[result.ordinal()].increment();
     super.end(result);
   }
 
   @Override
   public String toString() {
-    return counts.toString();
+    return Arrays.toString(counts);
```

Comment:
This doesn't product quite the same output... you'll get the counts in order but no printing of the associated keys.  Could we add the enum toString in to the output as well?

Reply:
Right. Done. Plus a test added.